sound/soc/sof/Kconfig

```diff
@@ -46,6 +46,15 @@ config SND_SOC_SOF_OF
 	  required to enable i.MX8 devices.
 	  Say Y if you need this option. If unsure select "N".
 
+config SND_SOC_SOF_COMPRESS
+	bool "SOF enable compress interface"
+	select SND_SOC_COMPRESS
```

Comment:
this will result in the SND_SOC_COMPRESS code being compiled as built-in, not really desirable.

I wonder if this should actually be a user-visible Kconfig option, should it not be selected by the platform instead. E.g. for IMX platform you select this option, but for Intel Baytrail you don't?

Reply:
We already have this:

```
config SND_SOC_COMPRESS                                                                                                                                                                
»       bool
»       select SND_COMPRESS_OFFLOAD
```

So it means that SND_SOC_COMPRESS is either builtin or not enabled. There is no M option.

You say that SND_SOC_COMPRESS should be user-visible Kconfig option right? 
Not sure if it makes any difference. 

With this patch, SND_SOC_SOF_COMPRESS is a user-visible Kconfig option and each platform can select it or not which in turn will compile SND_SOC_COMPRESS in.

Reply:
That's a bug at the core level then, the Makefile is inconsistent IMHO.

````
snd-soc-core-$(CONFIG_SND_SOC_COMPRESS) += soc-compress.o

ifneq ($(CONFIG_SND_SOC_TOPOLOGY),)
snd-soc-core-objs += soc-topology.o
endif
````

This should be 
````
ifneq ($(CONFIG_SND_SOC_TOPOLOGY),)
snd-soc-core-objs += soc-compress.o
endif
````

Reply:
never mind, this is not a bug but me not understanding how Kconfig works. when SND_SOC_COMPRESS=y SND_COMPRESS_OFFLOAD=m because SND_SOC=m

I still don't get the point. Either the user enables this option, and somewhere you have to use 'depends on' to test the value, or the platform makes the choice directly. Doing both is confusing, the user will NOT be able to de-select something that's already set, or the user will select something that was never needed.

Reply:
Can we have this option as non user visible and selected by platforms where the compressed is implemented?

Reply:
@ujfalusi We modified the code for this to be non user-visible.